### PR TITLE
Added Unit tests

### PR DESCRIPTION
**Pull request by SubtleDragon057**
Added editor Unit Tests for the 'Core' folder

### CODE REVIEW

File: Assets/Editor/Tests/EditMode/Core_Tests/DataManager_Tests/Connection_Tests/ConnectionDataManager_Tests.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+using AssemblyCSharp.Assets.Code.Core.DataManager.Impl.Connection;
+using AssemblyCSharp.Assets.Code.Core.DataManager.Interface.Connection.Entities;
+using AssemblyCSharp.Assets.Code.Core.Storage.Impl.Connection;
+using AssemblyCSharp.Assets.Code.Core.Storage.Impl.Providers.PlayerPrefs.Impl;
+using NUnit.Framework;
+
+public class ConnectionDataManager_Tests
```
Comment (BramDC3) [inline Assets/Editor/Tests/EditMode/Core_Tests/DataManager_Tests/Connection_Tests/ConnectionDataManager_Tests.cs:7]: File name and class name can just be ConnectionDataManagerTests I think?

**Comment by BramDC3** [comment]
I'm really glad that you got it to work! Something that's missing here is a mocking library though, so you don't have to create all necessary dependencies yourself.

File: Assets/Editor/Tests/EditMode/Core_Tests/DataManager_Tests/DataManager_Tests.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+using AssemblyCSharp.Assets.Code.Core.DataManager.Interface.Connection.Entities;
+using AssemblyCSharp.Assets.Code.Core.Storage.Impl.Connection;
+using AssemblyCSharp.Assets.Code.Core.Storage.Impl.Providers.PlayerPrefs.Impl;
+using AssemblyCSharp.Assets.Code.Core.Storage.Impl.Texture;
+using AssemblyCSharp.Assets.Code.Core.Storage.Interface.Texture;
+using AssemblyCSharp.Assets.Code.Core.YGOProDeck.Interface;
+using NSubstitute;
+using NUnit.Framework;
+
+public class DataManager_Tests
```
Comment (BramDC3) [inline Assets/Editor/Tests/EditMode/Core_Tests/DataManager_Tests/DataManager_Tests.cs:14]: No need to write tests for the data manager itself, only fur the sub data managers like the connection data manager etc.

File: Assets/Editor/Tests/EditMode/Core_Tests/General_Tests/Extensions_Tests/FloatExtensions_Tests.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+using AssemblyCSharp.Assets.Code.Core.General.Extensions;
+
+public class FloatExtensions_Tests
+{
+    private float testNumber = 10f;
+    
+    [Test]
+    public void Given_AFloatValue_When_TheValueIsWithinTheRange_Then_ReturnsTrue()
+    {
+        bool test = testNumber.IsWithinRange(1, 20);
```
Comment (BramDC3) [inline Assets/Editor/Tests/EditMode/Core_Tests/General_Tests/Extensions_Tests/FloatExtensions_Tests.cs:11]: Always good to test the edge cases with these tests. Values like -1, 0, 1, 20, 21, null would make good tests for this function. It's possible to only write 1 test and pass parameters to, more info here: https://www.lambdatest.com/blog/nunit-parameterized-test-examples/
